includes/collections/class-query-helper.php

```diff
@@ -0,0 +1,372 @@
+<?php
+/**
+ * Query helper class.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Collections;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Query helper class for Collections data operations and business logic.
+ */
+class Query_Helper {
+
+	public const COVER_SECTION = 'cover';
+	public const NO_SECTION    = 'no-section';
+
+	/**
+	 * Get available years from published collections for filtering.
+	 *
+	 * @param string $selected_category Optional category filter.
+	 * @return array Array of years.
+	 */
+	public static function get_available_years( $selected_category = '' ) {
+		$args = [
+			'post_type'      => Post_Type::get_post_type(),
+			'post_status'    => 'publish',
+			'posts_per_page' => -1,
+			'fields'         => 'ids',
+		];
+
+		// Add category filter if specified.
+		if ( ! empty( $selected_category ) ) {
+			$args['tax_query'] = [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+				[
+					'taxonomy' => Collection_Category_Taxonomy::get_taxonomy(),
+					'field'    => 'slug',
+					'terms'    => $selected_category,
+				],
+			];
+		}
+
+		$collections = get_posts( $args );
+		$years       = [];
+
+		foreach ( $collections as $collection_id ) {
+			$year = get_the_date( 'Y', $collection_id );
```

Comment:
I think we can find a more performant alternative for this method... it might involve building a direct SQL query but will be way faster and simpler than doing this. I think there's a way to do this with one single SQL query.